knora/dsplib/utils/expand_all_lists.py

```diff
@@ -1,37 +1,38 @@
-from typing import Any
+from typing import Any, Union
 
 from knora.dsplib.utils.excel_to_json_lists import make_json_list_from_excel, prepare_list_creation
 
 
-def expand_lists_from_excel(data_model: dict[str, Any]) -> list[str]:
+def expand_lists_from_excel(
+    data_model: dict[str, dict[str, list[dict[str, Union[str, dict[str, Any]]]]]]
+) -> list[dict[str, Any]]:
     """
-    Gets all list definitions from a data model and expands them to JSON if they are only referenced via an Excel file
+    Get all list definitions from a data model and expand them to JSON if they are only referenced via an Excel file
 
     Args:
         data_model: The data model (JSON) the lists are read from
 
     Returns:
         A list of all expanded lists. It can be added to the root node of an ontology as lists section.
     """
-    # create and add lists from Excel references to the ontology
-    lists = data_model["project"].get("lists")
 
-    if not lists:
+    if 'project' not in data_model or 'lists' not in data_model['project']:
         return []
 
+    lists = data_model['project']['lists']
```

Comment:
I see, but I like to have a clear separation between all tests (the extensive testing on line 19 you wondered about) and the straight access (line 22). 

If I use the construction with `.get()` thoroughly, it would need to be `lists = data_model.get('project').get('lists')`. But then, the first get could return `None`.

Reply:
Yes that's true, you would have to take it apart into two operations.